### PR TITLE
US17046 - Dynamic JSON Files

### PR DESCRIPTION
**Pull request by seancdavis**

### CODE REVIEW

File: src/components/shared-header/shared-header.tsx [hunk 1]
```diff
@@ -1,53 +1,34 @@
...
-   */
-  private getPayload() {
-    this.console.log('getRecords()');
-    this.payload = dig(window, 'CRDS', 'navigation') || [];
-    if (this.payload.length > 0 && this.src) {
-      axios.get(this.src).then(success => {
-        this.payload = dig(success, 'data');
-      });
-    }
+  fetchUrl() {
```
Comment (tcmacdonald) [inline src/components/shared-header/shared-header.tsx]: Can we move this method in to `shared/utils`?

File: src/components/shared-footer/shared-footer.tsx [hunk 2]
```diff
@@ -1,70 +1,69 @@
...
-      else {
-        if(el["automation-id"]){
-          return <a target={target} href={el.path} data-automation-id={el["automation-id"]} >{el.title}</a>
-        }
-        return <a target={target} href={el.path}>{el.title}</a>
-      }
-    } else {
-      return el.title
-    }
+  fetchUrl() {
```
Comment (tcmacdonald) [inline src/components/shared-footer/shared-footer.tsx]: Can we move this method in to `shared/utils`?

File: src/components/snail-trail/snail-trail.tsx [hunk 3]
```diff
@@ -1,43 +1,46 @@
...
+  @Prop() env: string = 'prod';
+  @Prop() name: string;
+
+  @State() data: Array<any> = [];
+
+  componentWillLoad() {
+    axios.get(this.fetchUrl()).then(response => (this.data = response.data));
+  }
+
+  fetchUrl() {
```
Comment (tcmacdonald) [inline src/components/snail-trail/snail-trail.tsx]: Can we move this method in to `shared/utils`?

File: src/index.html [hunk 4]
```diff
@@ -31,9 +31,16 @@
     <script src="/build/crds-components.js"></script>
   </head>
   <body>
-    <shared-header src="/data/navigation.json"></shared-header>
 
-    <div class="jumbotron jumbotron-xl" style="margin-top: 92px; background-image: url('//crds-cms-uploads.imgix.net/content/images/streaming-anywhere-hero-off.jpg?auto=format,compress&amp;w=100&amp;fit=crop');">
+    <div class="shared-header-skeleton" style="min-height: 49px; background-color: #0095d9">
```
Comment (tcmacdonald) [inline src/index.html]: Should these inline styles get folded into `crds-styles`?

File: src/components/shared-header/shared-header.tsx (hunk 1 above)
Comment (seancdavis) [reply src/components/shared-header/shared-header.tsx]: I could probably DRY this up more than I did, but I'll leave the rest for a refactoring task.

File: src/index.html (hunk 4 above)
Comment (seancdavis) [reply src/index.html]: Yes. Going to create a new story for this.

**Pull request merged**